Controllers/CHumidityController.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * CHumidityController.h
+ *
+ *  Created on: Aug. 11, 2022
+ *      Author: samk
+ */
+
+#ifndef CHUMIDITYCONTROLLER_H_
+#define CHUMIDITYCONTROLLER_H_
+
+#include "CBME280.h"
+#include "CController.h"
+#include "CPIDLoop.h"
+#include "IHardwareMap.h"
+
+#define CHANNEL_NUMBER 8
+
+class CHumidityController : public CController
+{
+public:
+    CHumidityController(IHardwareMap *p_hardwaremap,
+                        etl::string<MAX_STRING_SIZE> name,
+                        uint32_t run_period_ms,
+                        SPI_HandleTypeDef *p_spi,
+                        GPIO_TypeDef *p_slave_select_port,
+                        uint16_t slave_select_pin);
+    virtual void run();
+    virtual bool newCommand(ICommand *p_command, IComChannel *p_comchannel);
+    virtual void reset();
+
+private:
+    void sendStatus(IComChannel *p_comchannel);
+    ICommand::command_error_code_t setHumidity(ICommand *p_command);
+    ICommand::command_error_code_t overrideHeater(ICommand *p_command);
```

Comment:
I see you made an assumption that we are going to use heater as a humidifier. For now this is acceptable, but since part of the goal here is to educate you I thought I'd be appropriate to demonstrate how this would be working in a more advanced object oriented design.
Since we right now don't know what humidifier we are going to use, a more flexible design decision would have been to have another class
`CHumidifier` with methods such as `CHumidifier::startEvaporation()` and `CHumidifier::stopEvaporation()` or something along those lines. 
Having said that, your design is fairly easy to adapt to having external humidifier and we will perhaps do that once this bit is done.